scipy-1.0/paper.tex

```diff
@@ -185,6 +186,114 @@ \section*{Project organisation and community}
+around 76\% for the last six months (Figure~\ref{fig:coverage}),
+but some of the historical components of the code base
+would still benefit from increased test coverage. We are still striving
+to increase the number of high-quality SciPy usage tutorials beyond
+our current 15 section offering\cite{SciPy_tutorials}, but our
+standard documentation for library features is already in excellent shape.
+
+\begin{figure}[H]
+\centering
+\includegraphics[width=\textwidth]{static/coverage-chart}
+\caption{\% coverage of the SciPy code base by unit tests over the six
+	 months preceding June 28, 2018 as reported by the \texttt{codecov}
+	 service. The analysis of lines of source code covered is performed
+	 automatically in our continuous integration suite using the
+	 \texttt{pytest-cov} and \texttt{gcov} libraries.}
+\label{fig:coverage}
+\end{figure}
+
+The low-level Cython code in our library (which interacts with C-level
+code and exposes it for Python usage) could use some measure of modernization,
+including migration to typed memoryviews to handle NumPy arrays instead of
+the older syntax:
+
+\begin{lstlisting}[language=Python]
+# example of a function that takes a single
+# 3 dimensional typed memoryview as an argument
+# this allows Cython-level handling of a NumPy
+# array object
+cpdef int func_new(int[:, :, :] arr):
+    pass
+
+# example of the old syntax for handling
+# the same scenario
+cpdef int func_old(object[int, ndim=3, mode='strided'] arr):
+    pass
+
+\end{lstlisting}
+
+\texttt{fftpack} has some single-precision floating point errors, a caching bug,
+and would benefit from the addition of the Bluestein algorithm. There's also
+too much community code duplication with \texttt{numpy.fft}, which
+requires clarification of best choices and long-term plans.
+
+\texttt{scipy.integrate} has a new ODE solver interface, and we should aim
+to gracefully deprecate the older API. Support for integrating complex-valued
+functions and integrating multiple intervals could be beneficial.
+
+\texttt{scipy.interpolate} could benefit from a set of new features including
+spline fitting routines with better user control; integration, differentiation,
+and arithmetic routines for splines; transparent tensor-product splines,
+Non-Uniform Rational B-Splines support; and mesh refinement and coarsening of
+B-splines and corresponding tensor products.
+
+\texttt{scipy.io} would benefit from PCM float support, but we otherwise
+direct users with more sophisticated needs to domain-specific audio
+libraries.
+
+For \texttt{scipy.linalg}, we need to remove functions that are duplicate with 
+\texttt{numpy.linalg} unless performance is superior in SciPy,
+\texttt{get\_lapack\_funcs} should always use flapack, and
+we should continue to wrap more LAPACK functions. Ideally, we'd also
+add type-generic wrappers in the Cython BLAS and LAPACK interfaces, and
+transform many of the linear algebra routines into gufuncs.
```

Comment:
let's remove the `flapack` stuff, not very user relevant. The type-generic wrappers thing we could keep here, but only if we rephrase it so that it makes sense to the average reader

Reply:
The most important thing for `linalg` I think is to start upgrading to more recent LAPACK versions.

Reply:
I think for the average reader it is more important to mention that they can access the LAPACK funcs via either `get_lapack_funcs` or via `sp.linalg.lapack.<whatever>`.